base/src/main/scala/clazz/Applicative.scala

```diff
@@ -2,8 +2,16 @@ package scato
 package clazz
 
 abstract class Applicative[F[_]] {
-  def apply: Apply[F]
+
   def pure[A](a: A): F[A]
+  def ap[A, B](fa: F[A])(f: F[A => B]): F[B]
+
+  def map[A, B](ma: F[A])(f: (A) => B): F[B] = ap(ma)(pure(f))
+
+  final val apply: Apply[F] = new Apply[F] {
```

Comment:
the only case where you would give a reference is when you have an Apply[F] instance that you would want to promote to an Applicative[F]. But then you should have created an Applicative[F] in the first place, no?
Or is there another case?